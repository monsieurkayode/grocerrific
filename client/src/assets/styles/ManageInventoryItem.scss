.inventory__form {
  margin: 20px 0;

  .form-field {
    margin: 25px 0;

    span,
    .error {
      color: tomato;
    }

    .error {
      font-size: 14px;
      padding-top: 5px;
      font-family: $font-family-prompt;
    }

    input[type="text"],
    input[type="number"] {
      display: inline-block;
      width: 100%;
      padding: 5px 0 5px 10px;
      height: 30px;
      border-radius: 5px;
      border: 1px solid $light-grey-1;
      font-family: $font-family-roboto;
      font-size: 14px;
      outline: none;

      &:focus {
        border: 1px solid #4d75a1;
      }
    }

    input[type="text"],
    input[type="number"] {
      &.alert {
        border: 1px solid tomato;
      }
    }

    input[type="text"] {
      width: 100%;
      text-transform: capitalize;
    }

    .inline__field {
      display: inline-block;
      width: 45%;
    }

    .bottom {
      float: right;
    }


    label {
      display: block;
      margin-bottom: 10px;
      font-family: $font-family-josefin;
    }
  }
}