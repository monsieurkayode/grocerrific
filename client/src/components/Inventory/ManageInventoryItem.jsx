import React from 'react';
import {
  func, shape, string, bool
} from 'prop-types';

import Modal from '../common/Modal';
import TextField from '../common/TextField';

const ManageInventoryItem = ({
  title,
  closeModal,
  groceryItem,
  handleInputChange,
  handleSubmit,
  handleFocus,
  errors,
  saving
}) => (
  <Modal
    title={`${title} Grocery`}
    closeModal={closeModal}
  >
    <form className="inventory__form" onSubmit={handleSubmit}>
      <TextField
        name="name"
        label="Name"
        placeholder="Tomato"
        error={errors.name}
        value={groceryItem.name}
        onChange={handleInputChange}
        onFocus={handleFocus}
      />

      <div className="form-field bottom">
        <TextField
          name="price"
          label="Price"
          placeholder="345"
          type="number"
          error={errors.price}
          value={groceryItem.price}
          field="inline__field"
          onChange={handleInputChange}
          onFocus={handleFocus}
        />

        <TextField
          name="quantity"
          label="Quantity"
          placeholder="17"
          type="number"
          error={errors.quantity}
          value={groceryItem.quantity}
          field="inline__field"
          onChange={handleInputChange}
          onFocus={handleFocus}
        />

      </div>
      <button
        className="buy uppercase clear"
        type="submit"
        disabled={saving}
      >
        {saving ? 'Saving...' : title}
      </button>
    </form>
  </Modal>
);

ManageInventoryItem.propTypes = {
  closeModal: func.isRequired,
  handleInputChange: func.isRequired,
  groceryItem: shape({}).isRequired,
  title: string.isRequired,
  handleSubmit: func.isRequired,
  errors: shape({}).isRequired,
  handleFocus: func.isRequired,
  saving: bool.isRequired
};

export default ManageInventoryItem;
