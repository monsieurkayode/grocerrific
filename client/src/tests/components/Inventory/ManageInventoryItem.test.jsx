import React from 'react';
import { shallow } from 'enzyme';
// eslint-disable-next-line max-len
import ManageInventoryItem from '../../../components/Inventory/ManageInventoryItem';

const setup = () => {
  const props = {
    title: 'Add',
    groceryItem: {
      name: '',
      quantity: '',
      price: ''
    },
    closeModal: jest.fn(),
    handleSubmit: jest.fn(),
    handleFocus: jest.fn(),
    handleInputChange: jest.fn(),
    saving: false,
    errors: {}
  };
  const wrapper = shallow(<ManageInventoryItem {...props} />);

  return { wrapper, props };
};

describe('ManageInventoryItem Component', () => {
  const { wrapper, props } = setup();

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a form with three TextField children components', () => {
    expect(wrapper.find('form').length).toBe(1);
    expect(wrapper.find('TextField').length).toBe(3);
  });

  it('should call handleSubmit when the form is submitted', () => {
    const form = wrapper.find('form');
    form.simulate('submit');
    expect(props.handleSubmit).toHaveBeenCalled();
  });

  it('should display Add on the submit button when not saving', () => {
    const submitButton = wrapper.find('button');
    expect(submitButton.text()).toEqual('Add');
  });

  it('should display Saving... on the submit button while saving', () => {
    wrapper.setProps({ saving: true });
    const submitButton = wrapper.find('button');
    expect(submitButton.text()).toEqual('Saving...');
  });
});
