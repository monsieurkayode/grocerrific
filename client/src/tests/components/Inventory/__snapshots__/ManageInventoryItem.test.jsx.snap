// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ManageInventoryItem Component renders correctly 1`] = `
<Modal
  closeModal={[MockFunction]}
  title="Add Grocery"
>
  <form
    className="inventory__form"
    onSubmit={[MockFunction]}
  >
    <TextField
      error=""
      field="form-field"
      label="Name"
      name="name"
      onChange={[MockFunction]}
      onFocus={[MockFunction]}
      placeholder="Tomato"
      type="text"
      value=""
    />
    <div
      className="form-field bottom"
    >
      <TextField
        error=""
        field="inline__field"
        label="Price"
        name="price"
        onChange={[MockFunction]}
        onFocus={[MockFunction]}
        placeholder="345"
        type="number"
        value=""
      />
      <TextField
        error=""
        field="inline__field"
        label="Quantity"
        name="quantity"
        onChange={[MockFunction]}
        onFocus={[MockFunction]}
        placeholder="17"
        type="number"
        value=""
      />
    </div>
    <button
      className="buy uppercase clear"
      disabled={false}
      type="submit"
    >
      Add
    </button>
  </form>
</Modal>
`;
